language/diem-framework/modules/XDX.move

```diff
@@ -13,7 +13,7 @@ module XDX {
     use 0x1::DiemTimestamp;
 
     /// The type tag representing the `XDX` currency on-chain.
-    struct XDX has copy, drop, store { }
+    struct XDX has store { }
```

Comment:
This is an interesting one... Guessing the type system is upset if this doesn't have `store`, but it is also true that no instance of `XDX` ever appears in global storage.

Reply:
I could see this being a fun one to play with. I think it might also be interesting to think about what this means from a  compatibility perspective as well: any type with `store`  and/or `key` ability can't simply be removed or changed due to type resolution. But, deleting/changing a type without `store` or `key` abilities should be perfectly fine..

Reply:
This goes to the discussion about phantom types. Really we could have `struct XDX {}` if `struct Diem` could mark its `CoinType` as being phantom/ghost

Reply:
> But, deleting/changing a type without store or key abilities should be perfectly fine..

Changing should be ok, but (I think) not deleting since functions/structs in other modules may link against it?

Reply:
If we had private types it should then be okay. But I think you are right that even without store, you can't just remove the type. Maybe this is then okay to remove `store` from Events? Thoughts?

Reply:
Good point on not being able to remove the type. IMO I think it would be useful for events to still have `store` since if we use this "I should be able to change the structure of a non-store type without issue" as an initial test to determine if a type should have `store` then events fall into this category. Additionally even though events aren't stored in main storage they are being stored into a storage of some kind, so it feels weird for them to not have this. 

I don't have a super strong opinion about this, so if either you or Sam feel strongly that it shouldn't have `store` more than happy to go with that. 

Also as a note for the future, it may eventually be worth considering adding an `event` ability to capture this weirdness around events.

Reply:
I think another ability or not is going to be determined based on how we bring events closer into the system. 

I think the structure changing around store though is a good point. And I don't think that is something we want for events. I'll land with events having store